src/extension.ts

```diff
@@ -107,3 +144,133 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
+                if (downloadAppContent === 'true') {
+                    // tslint:disable-next-line: no-unsafe-any
+                    await extract(downloadFilePath, { dir: projectFilePath });
+                }
+                await requestUtils.downloadFile(
+                    `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
+                    vscode.Uri.joinPath(devContainerFolderPathUri, 'devcontainer.json').fsPath
+                );
+                await requestUtils.downloadFile(
+                    `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/Dockerfile`,
+                    vscode.Uri.joinPath(devContainerFolderPathUri, 'Dockerfile').fsPath
+                );
+            });
+            vscode.window.showInformationMessage(localize('restartingVsCodeInfoMessage', 'Restarting VS Code with your function app project'));
+            vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(projectFilePath));
+        } catch (err) {
+            vscode.window.showErrorMessage(localize('failedLocalProjSetupErrorMessage', 'Failed to set up your local project. Please try again.'));
+        } finally {
+            vscode.workspace.fs.delete(
+                vscode.Uri.file(toBeDeletedFolderPathUri.fsPath),
+                {
+                    recursive: true,
+                    useTrash: true
+                }
+            );
+        }
+    } else {
+        vscode.window.showErrorMessage(localize('invalidInputsErrorMessage', 'Invalid inputs. Please try again.'));
+    }
+}
+
+function getProjectLanguageForLanguage(language: string): ProjectLanguage {
+    switch (language) {
+        case 'powershell':
+            return ProjectLanguage.PowerShell;
+        case 'node':
+            return ProjectLanguage.TypeScript;
+        case 'python':
+            return ProjectLanguage.Python;
+        case 'java8':
+        case 'java11':
+            return ProjectLanguage.Java;
+        case 'dotnetcore2.1':
+        case 'dotnetcore3.1':
+            return ProjectLanguage.CSharp;
+        default:
+            vscode.window.showErrorMessage(localize('unsupportedLangErrorMessage', 'Language not supported: "{0}"', language));
+            throw new Error(`Language not supported: ${language}`);
+    }
+}
+
+// tslint:disable-next-line:no-any
+async function setupAzureAccount(account: AzureAccount): Promise<any> {
+    try {
+        await vscode.commands.executeCommand('azure-account.login');
+        return await account.sessions[0].credentials2.getToken();
+    } catch (err) {
+        vscode.window.showErrorMessage(localize('failedAzureAccSetupErrorMessage', 'Failed to setup Azure account. Please try again.'));
```

Comment:
@ejizba I am keeping this and the other comment open for now as I have clarifying questions. Since I am not able to use ext.tree.findTreeItem  as mentioned below, I need the account and token information for my current setup.

Reply:
Tried using ext.azureAccountTreeItem.getIsLoggedIn() but it only tells if I am logged in or not. How can I automatically prompt the login without using the logic around AzureAccount extension in case user is not logged in?

Reply:
Updated